File: Java/Array/program10.java
//Accept N numbers from user and check whether the enterd number is there in given numbers or not

import java.util.*;

class ArrayX
{
    public int Arr[];

    ArrayX(int iSize)
    {
        Arr = new int[iSize];
    }

    public void Accept()
    {
        int iCnt = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the elements of array");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public boolean Search(int iValue)
    {
        int iCnt = 0;
        boolean bFlag = false;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == iValue)
            {
                break;
            }
        }
        if(iCnt == Arr.length)
        {
            return false;
        }
        else
        {
            return true;
        }
    }
}

class program10
{
    public static void main(String arg[])
    {
        int iLength = 0;
        int iNo = 0;
        boolean bRet;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the size of array");
        iLength = sobj.nextInt();

        ArrayX obj = new ArrayX(iLength);

        obj.Accept();

        System.out.println("Enter the element to search");
        iNo = sobj.nextInt();

        bRet = obj.Search(iNo);
        if(bRet == true)
        {
            System.out.println("Element is there in array");
        }
        else
        {
            System.out.println("Element is not there in array");
        }
        
    }
}

File: C++/Array/program10.cpp
//Accept N numbers from user and check whether the enterd number is there in given numbers or not

#include<iostream>
using namespace std;

class ArrayX
{
    public:
        int *Arr;
        int iSize;

        ArrayX(int iLength)
        {
            iSize = iLength;
            Arr = new int[iSize];
        }

        ~ArrayX()
        {
            delete [] Arr;
        }

        void Accept()
        {
            cout<<"Enter the elements : "<<endl;
            for(int iCnt = 0; iCnt < iSize; iCnt++)
            {
                cin>>Arr[iCnt];
            }
        }

        bool Search(int iValue)
        {
            int iCnt = 0;

            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                if(Arr[iCnt] == iValue)
                {
                    break;
                }
            }
            if(iCnt == iSize)
            {
                return false;
            }
            else
            {
                return true;
            }
        }
};

int main()
{
    int iLength = 0;
    int iNo = 0;
    bool bRet;

    cout<<"Enter the size of array"<<endl;
    cin>>iLength;

    ArrayX obj1(iLength);

    obj1.Accept();

    printf("Enter the element to search\n");
    scanf("%d",&iNo);

    bRet = obj1.Search(iNo);
    if(bRet == true)
    {
        cout<<"Element is there in array"<<endl;
    }
    else
    {
        cout<<"Element is not there in array"<<endl;
    }
    return 0;
}